### PR TITLE
UI : Restricted special character for Glossary and Term

### PR DESCRIPTION
**Pull request by Ashish8689**
### Describe your changes :

- Issue : #9151
- Restricted special character for Glossary and Term
- Only Alpha Numeric with space is allowed

#
### Type of change :

- [x] Bug fix
- [x] Improvement

#
### Frontend Preview (Screenshots) :
image
image
image

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
 @open-metadata/ui

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/AddGlossary/AddGlossary.component.tsx [hunk 1]
```diff
@@ -108,7 +108,7 @@ const AddGlossary = ({
   const validateForm = () => {
     const errMsg = {
       name: !name.trim(),
-      invalidName: UrlEntityCharRegEx.test(name.trim()),
+      invalidName: specialCharacterRegEx.test(name),
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/AddGlossary/AddGlossary.component.tsx:111]: Shouldn't we trim the name?

File: openmetadata-ui/src/main/resources/ui/src/constants/regex.constants.ts [hunk 2]
```diff
@@ -25,3 +25,5 @@ export const nameWithSpace = /\s/g;
 
 export const passwordRegex =
   /^(?=.*\d)(?=.*[A-Z])(?=.*[a-z])(?=.*[^\w\d\s:])([^\s]){8,16}$/g;
+
+export const specialCharacterRegEx = /[`!@#$%^&*()+=[\]{};':"\\|,.<>/?~]/;
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/constants/regex.constants.ts:29]: Let's call it allowedNameRegEx for easy understanding

File: openmetadata-ui/src/main/resources/ui/src/components/AddGlossary/AddGlossary.component.tsx (hunk 1 above)
Comment (Ashish8689) [reply openmetadata-ui/src/main/resources/ui/src/components/AddGlossary/AddGlossary.component.tsx:111]: @Sachin-chaurasiya Not necessary as we are allowing the space and the regex neglect it.

**Comment by Ashish8689** [comment]
@chirag-madlani @Sachin-chaurasiya Change updated!

**Review by chirag-madlani — APPROVED** [review]

**Review by Sachin-chaurasiya — APPROVED** [review]

**Review by Sachin-chaurasiya — APPROVED** [review]

**Pull request merged**